saleor/order/actions.py

```diff
@@ -177,18 +179,19 @@ def cancel_order(
 
     Release allocation of unfulfilled order items.
     """
+    # transaction ensures proper allocation and event triggering
+    with traced_atomic_transaction():
+        events.order_canceled_event(order=order, user=user, app=app)
+        deallocate_stock_for_order(order, manager)
+        order.status = OrderStatus.CANCELED
+        order.save(update_fields=["status", "updated_at"])
 
-    events.order_canceled_event(order=order, user=user, app=app)
-    deallocate_stock_for_order(order, manager)
-    order.status = OrderStatus.CANCELED
-    order.save(update_fields=["status", "updated_at"])
-
-    transaction.on_commit(lambda: manager.order_cancelled(order))
-    transaction.on_commit(lambda: manager.order_updated(order))
+        call_event(manager.order_cancelled, order)
+        call_event(manager.order_updated, order)
 
-    transaction.on_commit(
-        lambda: send_order_canceled_confirmation(order, user, app, manager)
-    )
+        transaction.on_commit(
```

Comment:
Yes, after latest changes `call_event` takes function object and *args to either run the function with args normally or `on_commit ` . So it can be used with any function that needs to be wrapped in `transaction.on_commit`

Reply:
This will be a part of separate PR

Reply:
Also it could have more accurate name then e.g. `on_transaction_commit`, as it's not only about webhook events.